src/item.cpp

```diff
@@ -1260,6 +1262,98 @@ static void insert_separation_line( std::vector<iteminfo> &info )
     }
 }
 
+double item::effective_dps( const player &guy, monster &mon ) const
+{
+    const float mon_dodge = mon.get_dodge() * 5.0f;
+    float base_hit = guy.get_hit_base() + guy.get_hit_weapon( *this );
+    base_hit *= std::max( 0.25f, 1.0f - guy.encumb( bp_torso ) / 100.0f );
+    float mon_defense = mon_dodge + mon.size_melee_penalty();
+    double num_hits = 0;
+    constexpr double hit_trials = 1000.0;
```

Comment:
I feel like `hit_trials` ought to be an integer type, but it's not a big problem.

Reply:
Legacy of the original implementation, but I don't think it matters either.